brave/src/main/java/brave/internal/codec/EntrySplitter.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.codec;
+
+import brave.internal.Platform;
+
+/**
+ * Splits a character sequence that's in a delimited string trimming optional whitespace (OWS)
+ * before or after delimiters.
+ *
+ * <p>This is intended to be initialized as a constant, as doing so per-request will add
+ * unnecessary overhead.
+ */
+public final class EntrySplitter {
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+    int maxEntries = Integer.MAX_VALUE;
+    char entrySeparator = ',', keyValueSeparator = '=';
+    boolean trimOWSAroundEntrySeparator = true, trimOWSAroundKeyValueSeparator = true;
+    boolean keyValueSeparatorRequired = true, shouldThrow = false;
+
+    public Builder maxEntries(int maxEntries) {
+      if (maxEntries == 0) throw new NullPointerException("maxEntries == 0");
```

Comment:
Does it look more sense to use `IllegalArgumentException` instead of `NullPointerException`?

Reply:
good eye :)